lib/fizzy/execute.cpp

```diff
@@ -196,6 +196,48 @@ Instance instantiate(const Module& module)
     return instance;
 }
 
+namespace
+{
+template <typename DstT, typename SrcT>
+DstT extend(SrcT in)
```

Comment:
Alternative implementation (a bit better because `static_cast` is only used when needed).

```
template <typename DstT, typename SrcT>
DstT extend(SrcT in)
{
    if constexpr (std::is_signed<SrcT>::value)
    {
        using SignedDstT = typename std::make_signed<DstT>::type;
        return static_cast<DstT>(SignedDstT{in});
    }
    else
        return DstT{in};
}
```